paymentapi/src/test/java/fi/hel/verkkokauppa/payment/service/refund/PaytrailRefundPaymentServiceUnitTests.java

```diff
@@ -0,0 +1,221 @@
+        refundDto.setNamespace(namespace);
+        // TODO refund dto values to createRefundPayment
+
+        refundDto.setOrderId(orderId);
+        refundDto.setPriceNet("10");
+        refundDto.setPriceVat("0");
+        refundDto.setPriceTotal("10");
+        refundDto.setCustomerEmail(UUID.randomUUID() + "@ambientia.fi");
+
+        refundAggregateDto.setRefund(refundDto);
+        ArrayList<RefundItemDto> refundItemDtos = new ArrayList<>();
+        RefundItemDto itemDto = new RefundItemDto();
+
+        String itemMerchantId = "itemMerchantId";
+        itemDto.setMerchantId(itemMerchantId);
+        refundItemDtos.add(itemDto);
+        refundAggregateDto.setItems(refundItemDtos);
+
+        OrderWrapper orderWrapper = new OrderWrapper();
+        OrderDto order = new OrderDto();
+        order.setType(OrderType.ORDER);
+        orderWrapper.setOrder(order);
+
+        PaymentDto paymentDto = new PaymentDto();
+
+        paymentDto.setShopInShopPayment(false);
+        paymentDto.setPaytrailTransactionId("mock-transaction-id");
+        String paymentMethod = "payment-method";
+        paymentDto.setPaymentMethod(paymentMethod);
+
+        dto.setPayment(paymentDto);
+        dto.setOrder(orderWrapper);
+        dto.setRefund(refundAggregateDto);
+
+
+        PaytrailPaymentContext mockPaymentContext = createMockPaytrailPaymentContext(namespace, itemMerchantId);
+
+        when(refundPaymentService.createRefundToPaytrailAndCreateRefundPayment(any())).thenCallRealMethod();
+
+        when(refundPaymentService.createPaytrailRefundContext(any(), any(), any())).thenReturn(
+                mockPaymentContext
+        );
+
+        when(paytrailPaymentClient.createRefund(any(), any(), any(), any())).thenCallRealMethod();
+
+        when(paytrailPaymentClient.createPaytrailClientFromPaymentContext(any())).thenCallRealMethod();
+
+        when(paytrailAuthClientFactory.getClient(any(), any())).thenReturn(paytrailClient);
+
+        when(refundPayloadConverter.convertToPayload(any(), any(), any())).thenCallRealMethod();
+
+        when(env.getRequiredProperty(any())).thenReturn("http://localhost:3000/");
+
+        CompletableFuture<Pair<PaytrailResponse, String>> responseFuture = new CompletableFuture<>();
+        PaytrailResponse paytrailResponse = new PaytrailResponse();
+        paytrailResponse.setValid(true);
+        Executors.newCachedThreadPool().submit(() -> {
+            Thread.sleep(500);
+            responseFuture.complete(new Pair<>(paytrailResponse, "{}"));
+            return null;
+        });
+
+        Mockito.when(paytrailClient.sendRequest(Mockito.any())).thenReturn(responseFuture.thenApply(result -> paytrailResponse));
+
+        when(paytrailClient.getBaseUrl()).thenReturn("http://localhost:3000/");
+
+        PaytrailRefundCreateResponse refundCreateResponse = new PaytrailRefundCreateResponse();
+        PaytrailRefundResponse refundResponse = new PaytrailRefundResponse();
+        refundResponse.setValid(true);
```

Comment:
Change refundResponse.isValid(true) -> refundCreateResponse.isValid(true), because of the inheritance fix in paytrail-java-lib.